src/textual/command.py

```diff
@@ -68,43 +68,47 @@ class CommandSourceHit(NamedTuple):
     match_display: RenderableType
     """The Rich renderable representation of the hit.
 
-    Ideally a [rich Text object][rich.text.Text] object or similar.
+    Ideally a [rich Text object][rich.text.Text] object.
     """
 
-    command: CommandPaletteCallable
+    command: CallbackType
     """The function to call when the command is chosen."""
 
-    command_text: str
+    text: str | None = None
     """The command text associated with the hit, as plain text.
 
     This is the text that will be placed into the `Input` field of the
-    [command palette][textual.command_palette.CommandPalette] when a
+    [command palette][textual.command.CommandPalette] when a
     selection is made.
     """
 
-    command_help: str | None = None
+    help: str | None = None
     """Optional help text for the command."""
 
     def __lt__(self, other: object) -> bool:
-        if isinstance(other, CommandSourceHit):
-            return self.match_value < other.match_value
+        if isinstance(other, Hit):
+            return self.score < other.score
         return NotImplemented
 
     def __eq__(self, other: object) -> bool:
-        if isinstance(other, CommandSourceHit):
-            return self.match_value == other.match_value
+        if isinstance(other, Hit):
+            return self.score == other.score
         return NotImplemented
 
+    def __post_init__(self) -> None:
+        if self.text is None:
+            self.text = str(self.match_display)
```

Comment:
Attributes of a `NamedTuple` are normally read-only. Is there something special about a `__post_init__` that allows this assignment?

Reply:
Strange. I made `text` allow None, and assumed I would have to populate it. But this code will never even run, so I guess thats already taken care of elsewhere.

Reply:
As mentioned elsewhere; the upshot of this is that the `text` that's intended for the `Input` ends up being `"None"` -- this will be even more important if/when we do the thing we were talking about a while back, where a command can complete into the `Input` and then require further parameters. This is why the distinction between the `match_display` (which goes into the dropdown, and can be any renderable type: highlighted text, a Rich Table, probably a Rich Pixels Pokemon image, etc...) and the `text` is important here.